### DISCUSSION TITLE
Formula sometimes not showing and inconsitent results

### DISCUSSION THREAD

**Discussion by SchittBull** [RafaelGB/obsidian-db-folder#828] [Q&A]
### Contact Details

hello@manikmoellers.de

### What happened?

# ...not showing
I use a formular column to ad multiple columns together like this to generate a Name.
``
${row.Name}_${row.ContainerType}_${row.Category}_${row.SubCategory}
``
Often times the resulting Name is not shown until I refresh by changing the formula slightly back and forth.

# ...inconsitent results
Also if a row has no value sometimes it gets green and sometimes I get an "undefined". For example in row 11 "SubCategory" is missing but instead of getting an undefined, the "-" between "Name" and "ContainerType" is deleted and the font is green. On row 14 its just _undefined in at the end.
![image](https://user-images.githubusercontent.com/53765158/219513442-e78612c4-a772-45c5-ba78-2b2e1ab15500.png)

I uploaded the folder I used
[Pipeline.zip](https://github.com/RafaelGB/obsidian-db-folder/files/10761880/Pipeline.zip)

### What platform were you using?

Desktop

### Version of the plugin

3.3.0

### Are you using the latest version of the plugin in your Obsidian vault?

- [X] I have verified that I am on the latest version

### Are you check if there is a similar issue?

- [X] I have verified that there are not similar issues

**Comment by RafaelGB** [comment]
Hi @SchittBull . what i think is happening:
row 11, 12 and 13 => Formula string renders markdown and `_<your string>_` renders _italic_ . Subcategory key exists but is empty

row 14,15 and 16 => Subcategory key does not exists

my proposed solution:
```javascript
`${row.Name}_${row.ContainerType}_${row.Category}${row.SubCategory ? "_"+row.SubCategory : ""}`
```

**Reply by SchittBull** [reply]
Thank you for your reply @RafaelGB. Can you tell me what this part does?:
```
? "_"+row.SubCategory : ""}`
```
I already tried putting  ` bevore and after but somehow it doesn't show anything when I do this.

**Reply by RafaelGB** [reply]
Its a ternary expression in javascript:
if row.SubCategory is null apply "_"+row.SubCategory, else apply an  empty string ""

the ` was just a typo of the markdown expression of the response
